src/RequestBodyParser.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Request\Body;
+
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Http\Header;
+use Yiisoft\Http\Status;
+use Yiisoft\Request\Body\Parser\JsonParser;
+
+final class RequestBodyParser implements MiddlewareInterface
+{
+    private ResponseFactoryInterface $responseFactory;
+    private ContainerInterface $container;
+
+    private array $parsers = [
+        'application/json' => JsonParser::class,
+    ];
+    private bool $badRequestResponse = true;
+
+    public function __construct(ResponseFactoryInterface $responseFactory, ContainerInterface $container)
+    {
+        $this->responseFactory = $responseFactory;
+        $this->container = $container;
+    }
+
+    public function withAddedParser(string $mimeType, string $parserClass): self
+    {
+        if (trim($mimeType) === '') {
+            throw new \InvalidArgumentException('The mime type cannot be an empty string.');
+        }
+        if ($parserClass === '') {
+            throw new \InvalidArgumentException('The parser class cannot be an empty string.');
+        }
+        if (strpos($mimeType, '/') === false) {
```

Comment:
Are you sure this check is needed here?

Reply:
[The check is valid](https://tools.ietf.org/html/rfc7231#section-3.1.1.1) but yes, not sure it adds any value here. I'd remove it.